src/common/include/SIRF/common/geometrical_info.h

```diff
@@ -0,0 +1,182 @@
+/*
+  CCP PETMR Synergistic Image Reconstruction Framework (SIRF)
+  Copyright 2018 Commonwealth Scientific and Industrial Research Organisation's
+  Australian eHealth Research Organisation
+
+  This is software developed for the Collaborative Computational
+  Project in Positron Emission Tomography and Magnetic Resonance imaging
+  (http://www.ccppetmr.ac.uk/).
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+  http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+#ifndef SIRF_GEOMETRICAL_INFO_TYPE
+#define SIRF_GEOMETRICAL_INFO_TYPE
+
+template <int num_dimensions, typename T>
+class tVector {
+private:
+	typedef T _vectT[num_dimensions];
+
+public:
+	inline T& operator[](const int d)
+		{ return vect[d]; }
+	inline T operator[](const int d) const
+		{ return vect[d]; }
+
+private:
+	_vectT vect;
+};
+
+
+template <int num_physical_dimensions, int num_index_dimensions>
+class GeometricalInfo {
+public:
+	typedef tVector<num_physical_dimensions, float>     Coordinate;
+	typedef tVector<num_index_dimensions, unsigned int> Index;
+	// Eventually something here like
+	// Coordinate transform_index_to_physical_point(Index)
+	// Index transform_physical_point_to_index(Coordinate)
+};
+
+
+template <int num_dimensions>
```

Comment:
name `Voxelised..` implies 3D to me. Also, it would work for any basis-function (not just voxels).

Would it be easy to support non-orthogonal vectors? If so, I'd call this `GridGeometricalInfo` or `LatticeGeometricalInfo` or something. In STIR it'd be `CartesianGrid...` but I'm not so sure why we stuck the `Cartesian` in there.

Reply:
Just remembered that STIR's `CartesianGrid...` supposes the lattice vectors are orthogonal